dubbo-cluster/src/main/java/org/apache/dubbo/rpc/cluster/support/ClusterUtils.java

```diff
@@ -50,71 +41,42 @@ private ClusterUtils() {
-        Map<String, String> map = new HashMap<String, String>();
         Map<String, String> remoteMap = remoteUrl.getParameters();
 
-        if (remoteMap != null && remoteMap.size() > 0) {
-            map.putAll(remoteMap);
-
-            // Remove configurations from provider, some items should be affected by provider.
-            map.remove(THREAD_NAME_KEY);
-            map.remove(DEFAULT_KEY_PREFIX + THREAD_NAME_KEY);
-
-            map.remove(THREADPOOL_KEY);
-            map.remove(DEFAULT_KEY_PREFIX + THREADPOOL_KEY);
-
-            map.remove(CORE_THREADS_KEY);
-            map.remove(DEFAULT_KEY_PREFIX + CORE_THREADS_KEY);
-
-            map.remove(THREADS_KEY);
-            map.remove(DEFAULT_KEY_PREFIX + THREADS_KEY);
-
-            map.remove(QUEUES_KEY);
-            map.remove(DEFAULT_KEY_PREFIX + QUEUES_KEY);
-
-            map.remove(ALIVE_KEY);
-            map.remove(DEFAULT_KEY_PREFIX + ALIVE_KEY);
-
-            map.remove(Constants.TRANSPORTER_KEY);
-            map.remove(DEFAULT_KEY_PREFIX + Constants.TRANSPORTER_KEY);
+        if (remoteMap == null || remoteMap.size() <= 0) {
+            return remoteUrl.addParameters(localMap);
         }
 
-        if (localMap != null && localMap.size() > 0) {
-            Map<String, String> copyOfLocalMap = new HashMap<>(localMap);
-
-            if(map.containsKey(GROUP_KEY)){
-                copyOfLocalMap.remove(GROUP_KEY);
-            }
-            if(map.containsKey(VERSION_KEY)){
-                copyOfLocalMap.remove(VERSION_KEY);
-            }
+        // Remove configurations from provider, some items should not being affected by provider.
+        remoteMap.remove(THREAD_NAME_KEY);
+        remoteMap.remove(THREADPOOL_KEY);
+        remoteMap.remove(CORE_THREADS_KEY);
+        remoteMap.remove(THREADS_KEY);
+        remoteMap.remove(QUEUES_KEY);
+        remoteMap.remove(ALIVE_KEY);
+        remoteMap.remove(TRANSPORTER_KEY);
 
-            copyOfLocalMap.remove(RELEASE_KEY);
-            copyOfLocalMap.remove(DUBBO_VERSION_KEY);
-            copyOfLocalMap.remove(METHODS_KEY);
-            copyOfLocalMap.remove(TIMESTAMP_KEY);
-            copyOfLocalMap.remove(TAG_KEY);
-
-            map.putAll(copyOfLocalMap);
-
-            map.put(REMOTE_APPLICATION_KEY, remoteMap.get(APPLICATION_KEY));
+        if (localMap != null && localMap.size() > 0) {
+            String remoteApplication = remoteMap.get(APPLICATION_KEY);
+            remoteMap.putAll(localMap);
+            remoteMap.put(REMOTE_APPLICATION_KEY, remoteApplication);
 
             // Combine filters and listeners on Provider and Consumer
             String remoteFilter = remoteMap.get(REFERENCE_FILTER_KEY);
-            String localFilter = copyOfLocalMap.get(REFERENCE_FILTER_KEY);
+            String localFilter = localMap.get(REFERENCE_FILTER_KEY);
             if (remoteFilter != null && remoteFilter.length() > 0
                     && localFilter != null && localFilter.length() > 0) {

```

Comment:
Use `StringUtils.isNotEmpty(remoteFilter) && StringUtils.isNotEmpty(localFilter)` is better ?